app/controllers/kalibro_configurations_controller.rb

```diff
@@ -46,9 +46,29 @@ def exists
   end
 
   def metric_configurations
-    if set_kalibro_configuration
-      respond_to do |format|
-        format.json { render json: {metric_configurations: @kalibro_configuration.metric_configurations} }
+    return unless set_kalibro_configuration
```

Comment:
No, the `set_kalibro_configuration` method will respond with a 422 and the error message already.
Which is still wrong btw, it should respond with a 404